tests/test_time.py

```diff
@@ -5,20 +5,129 @@
 import pytest
 import xarray as xr
 from s2spy.time import AdventCalendar
+from s2spy.time import CustomCalendar
 from s2spy.time import MonthlyCalendar
+from s2spy.time import Period
+from s2spy.time import PrecursorPeriod
+from s2spy.time import TargetPeriod
 from s2spy.time import WeeklyCalendar
 
 
-def interval(start, end):
+def interval(start, end, closed="right"):
     """Shorthand for more readable tests."""
-    return pd.Interval(pd.Timestamp(start), pd.Timestamp(end))
+    return pd.Interval(pd.Timestamp(start), pd.Timestamp(end), closed = closed)
+
+
+class TestPeriod:
+    """Test Period objects."""
+
+    def test_period(self):
+        period = Period(length=10, gap=10)
+        assert isinstance(period, Period)
+
+    def test_target_period(self):
+        target = TargetPeriod(20, 10)
+        assert isinstance(target, TargetPeriod)
+
+    def test_precursor_period(self):
+        precursor = PrecursorPeriod(10, -5)
+        assert isinstance(precursor, PrecursorPeriod)
```

Comment:
Wouldn't it be good to also test that the properties are saved correctly here?
E.g. `assert precursor.length = 10`. (We'll have to change that to TimeDeltas later but putting this in already could be a good idea).

Reply:
Good idea. Tests added.